### PR TITLE
Cover `ENABLE_L7_LB` in complexity tests

### LINKED ISSUE #23062: Cover `ENABLE_L7_LB` in complexity tests
We don't currently cover `ENABLE_L7_LB` in complexity tests today, so we run the risk of having complexity regressions (if it's not already the case) with that feature.

Fixing that is a matter of updating `bpf/Makefile` and `bpf/complexity-tests/`. This work was started in https://github.com/cilium/cilium/pull/21998 before.

### PR DESCRIPTION
**Pull request by Thearas**
Fixes: https://github.com/cilium/cilium/issues/23062

### CODE REVIEW

**Comment by pchaigno** [comment]
Hi :wave: Thanks for contributing this! It looks like there's a bunch of conflicts that will need to be solved before we can merge.

**Comment by Thearas** [comment]
> Hi 👋 Thanks for contributing this! It looks like there's a bunch of conflicts that will need to be solved before we can merge.

Done.

File: bpf/complexity-tests/419/bpf_host.txt
Comment (pchaigno) [inline bpf/complexity-tests/419/bpf_host.txt]: Why did you add new configs here instead of extending the existing ones?

File: bpf/complexity-tests/419/bpf_overlay.txt
Comment (pchaigno) [inline bpf/complexity-tests/419/bpf_overlay.txt]: That doesn't look intended.

**Review by pchaigno — CHANGES_REQUESTED** [review]
A couple of issues from the rebase.

File: bpf/complexity-tests/419/bpf_overlay.txt
Comment (Thearas) [reply bpf/complexity-tests/419/bpf_overlay.txt]: Sorry, I forgot to remove it.

**Review by pchaigno — APPROVED** [review]

**Comment by ti-mo** [comment]
@Thearas Sorry, looks like this will need another rebase. Will merge after 1.14 release.

**Comment by ldelossa** [comment]
@Thearas gentle ping. If you'd like to move this PR forward please rebase.

**Pull request closed without merging**
